src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -812,29 +812,11 @@ def _get_grant_permissions(graph_client, client_sp_object_id=None, query_filter=
     return grant_info
 
 
-def list_permissions(cmd, identifier):
-    # the important and hard part is to tell users which permissions have been granted.
-    # we will due diligence to dig out what matters
-
-    graph_client = _graph_client_factory(cmd.cli_ctx)
-
-    # first get the permission grant history
-    client_sp_object_id = _resolve_service_principal(graph_client.service_principals, identifier)
-    grant_permissions = _get_grant_permissions(graph_client, client_sp_object_id=client_sp_object_id)
-
-    # get original permissions required by the application, we will cross check the history
-    # and mark out granted ones
+def list_required_permissions(cmd, identifier):
+    # Returns the required API permissions (requiredResourceAccess)
     graph_client = _graph_client_factory(cmd.cli_ctx)
     application = show_application(graph_client.applications, identifier)
     permissions = application.required_resource_access
-    for p in permissions:
-        result = list(graph_client.service_principals.list(
-            filter="servicePrincipalNames/any(c:c eq '{}')".format(p.resource_app_id)))
-        expiry_time = 'N/A'
-        if result:
-            expiry_time = ', '.join([x.expiry_time for x in grant_permissions if
```

Comment:
I agree with you. The expiry time we retrieved is only for delegated permissions. We shouldn't fix it with application permisions.

When the app only has application permissions, I got empty `expiryTime`:

```
> az ad app permission list --id 46eb4122-bd2b-4f54-af7b-6d79b46ee31a
[
  {
    "additionalProperties": null,
    "expiryTime": "",
    "resourceAccess": [
      {
        "additionalProperties": null,
        "id": "9a5d68dd-52b0-4cc2-bd40-abcf44ac3a30",
        "type": "Role"
      }
    ],
    "resourceAppId": "00000003-0000-0000-c000-000000000000"
  }
]
```

However, I am not sure if this BREAKING CHANGE is necessary or if it can cause customers' script to fail. @achandmsft, do you think removing the misleading `expiryTime` is a good idea?

Reply:
If any script is depending on this value, it is a good thing that it will fail, as the value itself is already incorrect.

Reply:
As discussed offline, I will add this property back (for permissions which have been granted tenant-wide, but the value will be set to a time far in the future, to ensure it is not misinterpreted as being expired.

Reply:
Updated to leave `expiryTime` in the response (if and only if there exists at least one tenant-wide delegated permission grant for the API), but sets the value far in the future. This avoids the breaking change, so I also updated the PR description to remove the BREAKING CHANGE.